### DISCUSSION TITLE
Difference between 'capture windows to disk' and 'capture snapshot'?

### DISCUSSION THREAD

**Discussion by RedSnt** [kangyu-california/PersistentWindows#298] [Q&A]
I was talking to my brother about the difference and 'capture windows to disk' and 'capture snapshot' features, and in the end it turned out we were both confused about the difference. I'm of the implicit understanding that one saves the snapshot in memory while the other saves the snapshot on disk, and that's the difference, but is there any advantages to any of them? Obviously the snapshot in memory is not persistent between reboots, but is that it?

EDIT: Thinking about it, because saving to disk sounds safer in general, is the reason that there are two methods because saving to disk all the time would wear out SSD's as an example? Is that the consideration?

**Comment by kangyu-california** [comment]
PW continuously saves window layout history in DDR memory, a snapshot is just a timestamp in that history, so that user can accurately restore window layout (and Z-order) of any historic moment. Note that snapshot restore does not attempt to recover closed/killed window, only live windows are restored. Also the memory of all snapshots are lost upon PC reboot or PW restart/upgrade.

Capture to disk, on the other hand, saves only window layout (not including the Z-order info) to SSD/HDD in LiteDB format, so that closed window can be reopened after reboot. User should expect slower and inaccurate restore due to the limitation of fuzzy matching method used in restore from disk.
